aptos-move/framework/aptos-framework/sources/TokenV1.move

```diff
@@ -280,18 +297,83 @@ module AptosFramework::TokenV1 {
         collection: vector<u8>,
         name: vector<u8>,
         amount: u64,
+        serial_number: u64
     ) acquires TokenStore {
-        let token_id = create_token_id_raw(creators_address, collection, name);
+        let token_id = create_token_id_raw(creators_address, collection, name, serial_number);
         direct_transfer(sender, receiver, token_id, amount);
     }
 
     public(script) fun initialize_token_script(account: &signer) {
         initialize_token_store(account);
     }
 
-    //
-    // Public functions for holding tokens
-    //
+    /// mutate the token property and save the new property in TokenStore
+    /// if the token serial_number is 0, we will create a new serial number per each token and store the properties
+    /// if the token serial_number is not 0, we will just update the propertyMap
+    public fun mutate_token_properties(
+        account: &signer,
+        token_owner: address,
+        token_id: TokenId,
+        amount: u64,
+        keys: vector<String>,
+        values: vector<vector<u8>>,
+        types: vector<String>,
+    ) acquires Collections, TokenStore {
+        // TODO: mutate based on capability
+        assert!(Signer::address_of(account) == token_owner, ENO_MUTATE_CAPABILITY);
+        // validate if the properties is mutable
+        assert!(exists<Collections>(token_id.token_data_id.creator), ECOLLECTIONS_NOT_PUBLISHED);
+        let all_token_data = &mut borrow_global_mut<Collections>(
+            token_id.token_data_id.creator
+        ).token_data;
+        let token_data = Table::borrow_mut(all_token_data, token_id.token_data_id);
+
+        assert!(token_data.mutability_config.properties, EFIELD_NOT_MUTABLE);
+        let addr = Signer::address_of(account);
+        // check if the serial_number is 0 to determine if we need to update the serial_number
+        let serial_number = token_id.serial_number;
```

Comment:
nit: serial_number is only used once so this can be merged with the next line

Reply:
updated